paasta_tools/setup_marathon_job.py

```diff
@@ -400,17 +408,43 @@ def log_deploy_error(errormsg, level='event'):
         bounce_health_params
     )
 
+    if new_app_running:
+        protected_draining_tasks = set()
+        if new_app.instances < config['instances']:
+            client.scale_app(app_id=new_app.id, instances=config['instances'], force=True)
+        elif new_app.instances > config['instances']:
+            num_tasks_to_scale = len(new_app.tasks) - config['instances']
+            task_dict = get_old_happy_unhappy_draining_tasks_for_app(
+                new_app,
+                drain_method,
+                service,
+                nerve_ns,
+                bounce_health_params,
+            )
+            scaling_app_happy_tasks = list(task_dict['happy'])
+            scaling_app_unhappy_tasks = list(task_dict['unhappy'])
+            scaling_app_draining_tasks = list(task_dict['draining'])
+            tasks_to_move_draining = max(min(len(scaling_app_draining_tasks), num_tasks_to_scale), 0)
+            num_tasks_to_scale = num_tasks_to_scale - tasks_to_move_draining
+            tasks_to_move_unhappy = max(min(len(scaling_app_unhappy_tasks), num_tasks_to_scale), 0)
+            num_tasks_to_scale = num_tasks_to_scale - tasks_to_move_unhappy
+            tasks_to_move_happy = max(min(len(scaling_app_happy_tasks), num_tasks_to_scale), 0)
+            protected_draining_tasks.union(scaling_app_draining_tasks[:tasks_to_move_draining])
+            old_app_draining_tasks[new_app.id] = set(scaling_app_draining_tasks[:tasks_to_move_draining])
+            old_app_live_unhappy_tasks[new_app.id] = set(scaling_app_unhappy_tasks[:tasks_to_move_unhappy])
+            old_app_live_happy_tasks[new_app.id] = set(scaling_app_happy_tasks[:tasks_to_move_happy])
```

Comment:
I'd move this line and the line above it up to just after you define `tasks_to_move_unhappy` and `tasks_to_move_happy`. This would make the structure of this section a bit more apparent.

Some vertical whitespace would help, too.